addons/stock_account/tests/test_stock_valuation_layer_revaluation.py

```diff
@@ -237,3 +237,58 @@ def test_stock_valuation_layer_revaluation_fifo(self):
 
         credit_lines = [l for l in new_layer.account_move_id.line_ids if l.credit > 0]
         self.assertEqual(len(credit_lines), 1)
+
+    def test_multi_company_fifo_svl_negative_revaluation(self):
+        company1 = self.env.company
+        company2 = next((c for c in self.env.companies if c != company1), None)
+
+        # Create product with category valuation set to real_time for company1
+        fifo_category = self.env['product.category'].create({
+            'name': 'FIFO',
+            'property_cost_method': 'fifo',
+            'property_valuation': 'real_time'
+        })
+        # Modify valuation to manual_periodic for company2
+        fifo_category.with_company(company2).write({
+            'property_valuation': 'manual_periodic'
+        })
+
+        product = self.env['product.product'].create({
+            'name': 'Test Product',
+            'type': 'product',
+        })
+        product.product_tmpl_id.categ_id = fifo_category
+
+        # Create moves to revaluate for company1
+        self._make_in_move(product, 10, unit_cost=10, create_picking=True)
+        self._make_out_move(product, 15, create_picking=True)
+
+        move3 = self.env['stock.move'].create({
```

Comment:
You should create the move in the same query as the picking:
```suggestion
receipt = self.env['stock.picking'].create({
            'picking_type_id': self.picking_type_in.id,
            'location_id': self.supplier_location.id,
            'location_dest_id': self.stock_location.id,
            'move_ids': [Command.create({
                'name': 'test fifo',
                'product_id': product.id,
                'location_id': self.supplier_location.id,
                'location_dest_id': self.stock_location.id,
                'product_uom': self.uom_unit.id,
                'product_uom_qty': 10,
                'price_unit': 7,
            })]
        }).with_company(company2)
```

Note that I add the `.with_company(company2)` after the create so that it is correctly created with the company of the env (company1) and then read and process by `company2`.